Sources/PackageModel/ToolchainConfiguration.swift

```diff
@@ -96,21 +96,50 @@ extension ToolchainConfiguration {
         public var manifestAPI: AbsolutePath
         public var pluginAPI: AbsolutePath
 
-        public init(manifestAPI: AbsolutePath, pluginAPI: AbsolutePath) {
+        public var manifestAPIMinimumDeploymentTarget: PlatformVersion?
+        public var pluginAPIMinimumDeploymentTarget: PlatformVersion?
+
+        public var macOSPackageDescriptionPath: AbsolutePath {
+            if manifestAPI.extension == "framework" {
+                return manifestAPI.appending(component: "PackageDescription")
+            } else {
+                // note: this is not correct for all platforms, but we only actually use it on macOS.
+                return manifestAPI.appending(component: "libPackageDescription.dylib")
+            }
+        }
+
+        public var macOSPluginLibraryPath: AbsolutePath {
```

Comment:
should these be put these macOSXXX APIs behind platform condition (`#if os(macOS)`)?

Reply:
I think that would make sense, I was just replicating what the existing code did.

Reply:
I think have a little extensions that is marked macOS only would be nicer way to organize the code

Reply:
Actually, this sounded good, but I don't think it's possible since we need the stored properties for deployment target. So I think it needs to be use `#if os(macOS)` in the main type, not an extension.